src/plugins/typechecker/README.md

```diff
@@ -0,0 +1,56 @@
+- infos = Information about the typechecker plugin is in keys below
+- infos/author = e1528532 <e1528532@libelektra.org>
+- infos/licence = BSD
+- infos/needs = 
+- infos/provides = typechecker
+- infos/recommends =
+- infos/placements = postgetstorage presetstorage
+- infos/status = maintained experimental global
+- infos/metadata =
+- infos/description = a plugin which typechecks configuration specifications
+
+## Introduction
+
+A plugin which typechecks specifications before setting keys and after getting keys from
+a mounted specification. 
+
+## Usage
+
+In order to use the type checker, mount a configuration specification along with this
+plugin. 
+
+`kdb mount <specification> spec/<path> <storage plugin to read the specification> typechecker`
+
+Currently there is no way to check the specification automatically upon mounting. Simply 
+retrieve a random key from the specification using `kdb get <path>` will cause
+the typechecking to happen, issuing a warning if it detects any problem with it.
+
+## Examples
+
+To see its functionality there will be a test specification installed into 
+`/usr/local/share/elektra/specifications/simpleTestSpecification.ini`. This
+path may vary slightly depending on the build configuration. The file can be
+typechecked by mounting it along with the typechecker plugin by calling
+
+`kdb mount simpleTestSpecification.ini spec/examples ini typechecker`
+
```

Comment:
Please make shell recorder test. Run at at least once `kdb get` and `kdb set` so that some behavior is triggered.

Reply:
@markus2330 i've added a shell recorder test, at least it already works locally which should show how things currently work when editing specifications manually.

Reply:
Should I review this PR again? I am afraid I cannot do this before end of the week.

Reply:
@markus2330 yes please, in the meanwhile i'll still add some minor improvements but basically it should be ready for another round.

Reply:
please remove "work in progress" if finished. Please reping me if I did not look at it shortly after the release (hopefully in the next weekend).